core/store/src/db.rs

```diff
@@ -431,3 +464,125 @@ impl TestDB {
         Self { db: RwLock::new(db) }
     }
 }
+
+impl RocksDB {
+    /// ColState has refcounted values.
+    /// Merge adds refcounts, zero refcount becomes empty value.
+    /// Empty values get filtered by get methods, and removed by compaction.
+    fn refcount_merge(
+        _new_key: &[u8],
+        existing_val: Option<&[u8]>,
+        operands: &mut MergeOperands,
+    ) -> Option<Vec<u8>> {
+        let mut result = vec![];
+        if let Some(val) = existing_val {
+            // Error is only possible if decoding refcount fails (=value is between 1 and 3 bytes)
```

Comment:
why cannot value be between 1 and 3 bytes?

Reply:
we write (4 bytes of refcount + value), or empty to delete (because merge cannot delete).
1..3 is the only case when `merge_refcounted_records` can error, and it shouldn't be possible